### ISSUE TITLE
UI: "Create" menu for creating primitives with UV2 present in 3D scene

### ISSUE THREAD

**Issue by SubtleMetaphor** [godotengine/godot-proposals#4895]
### Describe the project you are working on

...

### Describe the problem or limitation you are having in your project

Most game engines have a command to immediately create primitives for use in the engine. In Godot this currently requires more clicks.

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

I propose a "Create" menu option to appear in the main toolbar when working in a 3D scene. This menu allows for creation of primitives.

The primitives should be created with UV2 present, meaning they can be used immediately in lightbaking. (This is currently not the case when creating primitives in Godot.)

![a58b11203ad238f44c84112193fbaca5](https://user-images.githubusercontent.com/6630276/179395917-ea8eec87-440a-4703-a7dc-13b11cda5b57.png)

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

...

### If this enhancement will not be used often, can it be worked around with a few lines of script?

...

### Is there a reason why this should be core and not an add-on in the asset library?

...

**Comment by Zireael07** [comment]
You can create primitives already. Adding UV2 to them was also suggested and IIRC it is worked on.

**Comment by SubtleMetaphor** [comment]
> You can create primitives already. Adding UV2 to them was also suggested and IIRC it is worked on.

Thanks for pointing that out @Zireael07. Currently, the only way I've found out to create primitives is to create a new mesh instance, then create a new primitive mesh in the mesh instance. Creating the actual resource itself requires 1 click for opening the node creation window, then typing for finding "MeshInstance", and then finally another press or click to create (alternatively, a double click on a "recent" or "favorite"). Then you need to click the mesh field of the mesh instance, and finally what kind of mesh you want.

Perhaps there is a faster way to do it, but this is the way I've found out so far. If there is a faster way, please educate me.

My proposal is a way to reduce the amount of clicks to achieve the same thing, plus make newcomers feel more at home (especially those from Unity), since these kinds of commands exist there. It's a way to reduce friction.

All the best.

**Comment by Zireael07** [comment]
No, there is no faster way. However, Godot is not a Unity clone, nor any other game engine clone. Toolbars already suffer from being overstuffed with things, and adding one more menu isn't the way to go.

(IIRC there was a suggestion of two to reduce the amount of clicks needed to create resources in general, but I can't find it)

**Comment by SubtleMetaphor** [comment]
> Toolbars already suffer from being overstuffed with things, and adding one more menu isn't the way to go.

There's only 5 toolbar options. I don't think that's overstuffed. Adding a create menu would not only increase productivity but also make newcomers feel more at home. In my opinion this outweighs any concerns of overstuffing the interface.

All the best.

**Comment by Zireael07** [comment]
Ah, sorry, didn't notice you added it to the top bar, next to the Help.

That menu placement doesn't make sense, as the feature you propose is only relevant for 3D scenes, so it'd have to be in the more ckuttered toolbar where Transform and Menu are. (Maybe placed under that Menu...)?

**Comment by SubtleMetaphor** [comment]
I think surely the editor can detect whether the current scene is a 2D or 3D one, and adapt the toolbar depending on which kind is active.

**Comment by Zireael07** [comment]
Yes, that already happens, but to the toolbar on top of the main viewport not to the bar where Help is.

**Comment by SubtleMetaphor** [comment]
Ah, you mean placing it under the scene tab, next to "Transform" or "View"?

![06bfa645ecc7aa36605b9f56443d055f](https://user-images.githubusercontent.com/6630276/179396772-9d2046dc-0234-4b00-8cc7-0784200d185f.png)

**Comment by Zireael07** [comment]
Yeah.

**Comment by SubtleMetaphor** [comment]
Yeah, that works too I think. 👍

**Comment by Zireael07** [comment]
More on issues with this bar I mentioned: https://github.com/godotengine/godot-proposals/issues/3550

That's why I don't expect this proposal will be acted on, at least not before that UI issue is fixed (it's a real annoyance to people using Android Godot or Godot on smaller screens)

**Comment by Calinou** [comment]
> The primitives should be created with UV2 present, meaning they can be used immediately in lightbaking. (This is currently not the case when creating primitives in Godot.)

This is being implemented by https://github.com/godotengine/godot/pull/49932, but I need help for adding UV padding to the meshes. The lightmap texel size also needs to be set according to the primitive mesh's size.

**Comment by Zireael07** [comment]
@Calinou to the rescue as always (*cough cough* do you ever sleep, Cal?)
